plugins/net_plugin/net_plugin.cpp

```diff
@@ -4417,6 +4457,23 @@ namespace eosio {
          } else {
             ++num_peers;
          }
+         net_plugin::p2p_per_connection_metrics::connection_metric metrics{
+              .address = (*it)->remote_endpoint_ip_array
+            , .port = (*it)->remote_endpoint_port
+            , .accepting_blocks = (*it)->is_blocks_connection()
+            , .last_received_block = (*it)->get_last_received_blk_num()
+            , .first_available_block = (*it)->get_peer_start_block_num()
+            , .last_available_block = (*it)->get_peer_head_block_num()
+            , .unique_first_block_count = (*it)->get_unique_blocks_rcvd_count()
+            , .latency = (*it)->get_peer_ping_time_ns()
+            , .bytes_received = (*it)->get_bytes_received()
+            , .last_bytes_received = (*it)->get_last_bytes_received()
+            , .bytes_sent = (*it)->get_bytes_sent()
+            , .last_bytes_sent = (*it)->get_last_bytes_sent()
+            , .connection_start_time = (*it)->connection_time
+            , .log_p2p_address = (*it)->log_p2p_address
+         };
+         per_connection.peers.push_back(metrics);
```

Comment:
Only do this work if `update_p2p_connection_metrics`.

Reply:
Added check of `update_p2p_connection_metrics`.